src/audio/eq_fir/eq_fir.c

```diff
@@ -800,8 +833,8 @@ static int eq_fir_reset(struct comp_dev *dev)
 
 	eq_fir_free_delaylines(cd);
 
-	cd->eq_fir_func_even = eq_fir_s32_passthrough;
-	cd->eq_fir_func = eq_fir_s32_passthrough;
+	cd->eq_fir_func_even = NULL;
+	cd->eq_fir_func = NULL;
```

Comment:
Does it work without these defaults?

Reply:
In eq_fir_prepare() is set appropriate fir function, corresponding to data format. Prepare must be called before eq_fir_copy(). To be certain I added assertion checking those functions callbacks before COMP_TRIGGER_START and COMP_TRIGGER_RELEASE commands.